src/modules/ejector/ejector.py

```diff
@@ -296,40 +272,49 @@ def _get_latest_exit_epoch(self, blockstamp: BlockStamp) -> tuple[EpochNumber, i
         """Returns amount of epochs that will take to sweep all validators in chain."""
-        validators = self.w3.cc.get_validators(blockstamp)
+        chain_config = self.get_chain_config(blockstamp)
+        total_withdrawable_validators = self._get_total_withdrawable_validators(blockstamp)
+
+        full_sweep_in_epochs = total_withdrawable_validators / MAX_WITHDRAWALS_PER_PAYLOAD / chain_config.slots_per_epoch
+        return int(full_sweep_in_epochs * self.AVG_EXPECTING_WITHDRAWALS_SWEEP_DURATION_MULTIPLIER)
 
-        total_withdrawable_validators = len(list(filter(lambda validator: (
+    def _get_total_withdrawable_validators(self, blockstamp: ReferenceBlockStamp) -> int:
+        total_withdrawable_validators = ilen(filter(lambda validator: (
             is_partially_withdrawable_validator(validator) or
             is_fully_withdrawable_validator(validator, blockstamp.ref_epoch)
-        ), validators)))
+        ), self.w3.cc.get_validators(blockstamp)))
 
-        chain_config = self.get_chain_config(blockstamp)
-        full_sweep_in_epochs = total_withdrawable_validators / MAX_WITHDRAWALS_PER_PAYLOAD / chain_config.slots_per_epoch
-        return int(full_sweep_in_epochs * self.AVG_EXPECTING_WITHDRAWALS_SWEEP_DURATION_MULTIPLIER)
+        logger.info({'msg': 'Calculate total withdrawable validators.', 'value': total_withdrawable_validators})
+        return total_withdrawable_validators
 
     @lru_cache(maxsize=1)
     def _get_churn_limit(self, blockstamp: ReferenceBlockStamp) -> int:
-        total_active_validators = reduce(
-            lambda total, validator: total + int(is_active_validator(validator, blockstamp.ref_epoch)),
-            self.w3.cc.get_validators(blockstamp),
-            0,
-        )
-        return max(MIN_PER_EPOCH_CHURN_LIMIT, total_active_validators // CHURN_LIMIT_QUOTIENT)
-
-    def _get_processing_state(self, blockstamp: BlockStamp) -> EjectorProcessingState:
-        ps = named_tuple_to_dataclass(
-            self.report_contract.functions.getProcessingState().call(block_identifier=blockstamp.block_hash),
-            EjectorProcessingState,
-        )
-        logger.info({'msg': 'Fetch processing state.', 'value': ps})
-        return ps
+        total_active_validators = self._get_total_active_validators(blockstamp)
+        churn_limit = compute_exit_churn_limit(total_active_validators)
+        logger.info({'msg': 'Calculate churn limit.', 'value': churn_limit})
+        return churn_limit
+
+    def _get_total_active_validators(self, blockstamp: ReferenceBlockStamp) -> int:
+        total_active_validators = len([
+            is_active_validator(val, blockstamp.ref_epoch)
+            for val in self.w3.cc.get_validators(blockstamp)
+        ])
+        logger.info({'msg': 'Calculate total active validators.', 'value': total_active_validators})
+        return total_active_validators
 
     def is_main_data_submitted(self, blockstamp: BlockStamp) -> bool:
```

Comment:
There are inconsistent logging style between modules, nice to have single pattern in all modules